hail/src/main/scala/is/hail/cxx/Emit.scala

```diff
@@ -861,6 +861,7 @@ class Emitter(fb: FunctionBuilder, nSpecialArgs: Int, ctx: SparkFunctionContext)
       case ir.NDArrayMap(child, elemName, body) =>
         val childTyp = child.pType.asInstanceOf[PNDArray]
         val elemPType = childTyp.elementType
+        val nDims = childTyp.nDims.asInstanceOf[PNat].n
```

Comment:
`Nat` definitely doesn't need a physical type.

Reply:
Cool, should `TNat` throw an unsupported exception for its `physicalType`?